### PR TITLE
[RFC] When resetting password check that requested user is not expired and/or disabled

### PR DESCRIPTION
**Pull request by stloyd**
I'm just not sure about naming etc. (for sure error message needs to be polished).

This PR should fix: #965

### CODE REVIEW

**Comment by sskorc** [comment]
:+1:

**Comment by stloyd** [comment]
@stof @lsmith77 rebased, ready for review ;)

**Comment by jacekelgda** [comment]
+1

**Comment by kedrap** [comment]
:+1:

**Comment by filipgolonka** [comment]
:+1:

**Comment by mplachta** [comment]
:+1:

**Comment by marphi** [comment]
:+1:

**Comment by duzymaju** [comment]
:+1:

**Comment by KamilWojciech** [comment]
:+1:

**Comment by wojteks27** [comment]
:+1:

**Comment by ulff** [comment]
:+1:

**Comment by sskorc** [comment]
@stof or @lsmith77, any chance for any feedback on this one?

**Comment by spike31** [comment]
:+1:

**Comment by laskos** [comment]
:+1:

**Comment by ErikTrapman** [comment]
Any updates on this one? There's quite some issues addressing this behaviour (#1295 #1413 ao).

**Comment by romaricdrigon** [comment]
:+1: 

This issue can have some pretty serious consequences, having expired users accessing your service is not really funny...

**Comment by victorhaggqvist** [comment]
+1

**Comment by yceruto** [comment]
:+1:

**Comment by patrickmatsumura** [comment]
👍

**Comment by rjd22** [comment]
@stof are you too busy to review of is something missing in this PR?

**Comment by sstok** [comment]
@rjd22 Please see my comment https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1020#issuecomment-26701662 accepting this pr as-is would introduce a security problem.

**Comment by XWB** [comment]
@stloyd Do you still have this PR somewhere? If so, can you rebase from master?

File: Controller/ResettingController.php [hunk 1]
```diff
@@ -94,26 +98,28 @@ public function checkEmailAction(Request $request)
...
-        /** @var $dispatcher \Symfony\Component\EventDispatcher\EventDispatcherInterface */
-        $dispatcher = $this->get('event_dispatcher');
 
         $user = $userManager->findUserByConfirmationToken($token);
 
         if (null === $user) {
             throw new NotFoundHttpException(sprintf('The user with "confirmation token" does not exist for value "%s"', $token));
         }
 
+        /** @var $dispatcher \Symfony\Component\EventDispatcher\EventDispatcherInterface */
```
Comment (OskarStark) [inline Controller/ResettingController.php]: please use use statement

File: Controller/ResettingController.php [hunk 2]
```diff
@@ -94,26 +98,28 @@ public function checkEmailAction(Request $request)
...
+        $dispatcher = $this->container->get('event_dispatcher');
+
         $event = new GetResponseUserEvent($user, $request);
         $dispatcher->dispatch(FOSUserEvents::RESETTING_RESET_INITIALIZE, $event);
 
         if (null !== $event->getResponse()) {
             return $event->getResponse();
         }
 
+        /** @var $formFactory \FOS\UserBundle\Form\Factory\FactoryInterface */
```
Comment (OskarStark) [inline Controller/ResettingController.php]: please use use statement

**Comment by OskarStark** [comment]
LGTM

**Comment by core23** [comment]
Already fixed in https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2076

**Pull request closed without merging**